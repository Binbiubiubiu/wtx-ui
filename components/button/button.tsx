import React, { MouseEventHandler, FC } from "react";
import cls from "classnames";

const defaultProps = {
  prefixCls: "wtx-button",
  up: false,
  down: false,
  actived: false,
  disabled: false
};

export interface ButtonProps {
  /**是否显示左边箭头
   *
   * @default false
   */
  up?: boolean;
  /**是否显示右边箭头
   *
   * @default false
   */
  down?: boolean;
  /**是否激活
   *
   * @default false
   */
  actived?: boolean;
  /**是否禁用
   *
   * @default false
   */
  disabled?: boolean;
  /**点击事件 */
  onClick?: MouseEventHandler<HTMLButtonElement>;
  [prop: string]: any;
}

export const Button: FC<ButtonProps> = ({
  prefixCls,
  up, // 左箭头
  down, // 右箭头
  actived, // 按钮激活
  disabled, // 按钮禁用
  onClick: handleClick, // 点击事件
  children,
  className,
  ...rest
}) => {
  return (
    <button
      className={cls(
        prefixCls,
        {
          [`${prefixCls}__up`]: up,
          [`${prefixCls}__down`]: down,
          [`${prefixCls}__actived`]: actived
        },
        className
      )}
      disabled={disabled}
      onClick={e => handleClick && !disabled && handleClick(e)}
      {...rest}
    >
      {children}
    </button>
  );
};

Button.defaultProps = defaultProps;

export default Button;
