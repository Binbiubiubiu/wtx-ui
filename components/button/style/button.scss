@import '../../theme/var/index.scss';
@import '../../theme/mixin.scss';

$prefixCls: '#{$prefixlib}button';

.#{$prefixCls} {
  background-color: $--button-primary-bg;
  border-radius: 24px;
  color: $--button-text-color;
  cursor: pointer;
  display: inline-block;
  font-size: $--button-text-size;
  line-height: 1;
  padding: 4px 12px;
  white-space: nowrap;

  &:active {
    background-color: darken($--button-primary-bg, 10);
  }

  &__actived {
    background-color: $--button-actived-bg;

    &:active {
      background-color: darken($--button-actived-bg, 10);
    }
  }

  &:disabled {
    background: $--button-disabled-bg;
    cursor: not-allowed;
  }

  &__up {
    background-image: url('../assets/up_arrow.png');
    background-position: left 8px center;
    background-repeat: no-repeat;
    background-size: 8px 14px;
    padding-left: 24px;
  }

  &__down {
    background-image: url('../assets/down_arrow.png');
    background-position: right 8px center;
    background-repeat: no-repeat;
    background-size: 8px 14px;
    padding-right: 24px;
  }
}
