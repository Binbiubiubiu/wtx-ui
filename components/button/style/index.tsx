import "../../theme";
import "./button.scss";
