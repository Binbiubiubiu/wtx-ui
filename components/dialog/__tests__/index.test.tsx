import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dialog from '../index';
import Icon from '../../icon';
import { Button } from '../../button';

describe('Dialog', () => {
  it('should render Dialog components', () => {
    const wrapper = render(
      <>
        <Dialog
          titleIcon={<Icon className="m-r-8" path="assets/Q017.png" width="16" height="18" />}
          title="视频分析"
          headerExpandRender={() => <Button>近七天记录</Button>}
          onClose={() => {}}
        >
          <div style={{ width: 100, height: 100 }} />
        </Dialog>
        <Dialog
          titleIcon={<Icon className="m-r-8" path="assets/Q017.png" width="16" height="18" />}
          theme="light"
          title="视频分析"
          headerExpandRender={() => <Button>近七天记录</Button>}
          onClose={() => {}}
        >
          <div style={{ width: 100, height: 100 }} />
        </Dialog>
      </>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should style output is not null', () => {
    expect(require('../style')).not.toBeNull();
  });
});
