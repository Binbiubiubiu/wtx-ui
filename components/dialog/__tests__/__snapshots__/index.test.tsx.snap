// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render Dialog components 1`] = `
Array [
  <section
    class="wtx-dialog__theme-dark"
  >
    <div
      class="wtx-dialog__header"
    >
      <img
        alt=""
        class="wtx-image-icon m-r-8"
        src="assets/Q017.png"
        style="width:16;height:18"
      />
      <span
        class="wtx-dialog__title"
      >
        视频分析
      </span>
      <div
        class="wtx-dialog__expand"
      >
        <button
          class="wtx-button"
          type="button"
        >
          近七天记录
        </button>
      </div>
      <img
        alt=""
        class="wtx-image-icon wtx-image-icon__clickable"
        src="/assets/close-icon.png"
        style="width:18;height:18"
      />
    </div>
    <div
      class="wtx-dialog__content"
    >
      <div
        style="width:100px;height:100px"
      />
    </div>
  </section>,
  <section
    class="wtx-dialog__theme-light"
  >
    <div
      class="wtx-dialog__header"
    >
      <img
        alt=""
        class="wtx-image-icon m-r-8"
        src="assets/Q017.png"
        style="width:16;height:18"
      />
      <span
        class="wtx-dialog__title"
      >
        视频分析
      </span>
      <div
        class="wtx-dialog__expand"
      >
        <button
          class="wtx-button"
          type="button"
        >
          近七天记录
        </button>
      </div>
      <img
        alt=""
        class="wtx-image-icon wtx-image-icon__clickable"
        src="/assets/close-icon.png"
        style="width:18;height:18"
      />
    </div>
    <div
      class="wtx-dialog__content"
    >
      <div
        style="width:100px;height:100px"
      />
    </div>
  </section>,
]
`;
